rinja_derive/src/generator.rs

```diff
@@ -1152,16 +1152,28 @@ impl<'a> Generator<'a> {
                 .or_insert_with(|| import.clone());
         }
 
+        let buf_writable = mem::take(&mut self.buf_writable);
         let mut child = Self::new(
             self.input,
             self.contexts,
             Some(heritage),
-            // Variables are NOT inherited from the parent scope.
-            MapChain::default(),
+            // FIXME: Fix the broken lifetimes.
+            //
+            // `transmute` is here to fix the lifetime nightmare around `&self.locals` not
+            // living long enough. The correct solution would be to make `MapChain` take two
+            // lifetimes `&'a MapChain<'a, 'b, K, V>`, making the `parent` field take
+            // `<'b, 'b, ...>`. Except... it doesn't work because `self` still doesn't live long
+            // enough here for some reason...
+            MapChain::with_parent(unsafe {
+                mem::transmute::<
```

Comment:
I'm still very unhappy about this. I was able to improve the situation a lot but still get this last error:

```
error: lifetime may not live long enough
    --> rinja_derive/src/generator.rs:1166:13
     |
84   | impl<'a, 'b> Generator<'a, 'b> {
     |      --  -- lifetime `'b` defined here
     |      |
     |      lifetime `'a` defined here
...
1166 |             MapChain::with_parent(&self.locals),
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ argument requires that `'b` must outlive `'a`
     |
     = help: consider adding the following bound: `'b: 'a`
```

I pushed my changes [here](https://github.com/rinja-rs/rinja/compare/master...GuillaumeGomez:rinja:reduce-lifetime-error?expand=1).

I'll try to continue to understand what's wrong. If anyone wants to give it a try though...